packages/elements-core/src/utils/ref-resolving/resolvedObject.ts

```diff
@@ -0,0 +1,73 @@
+import { isArray, isPlainObject } from 'lodash';
+
+import { defaultResolver, ReferenceResolver } from './ReferenceResolver';
+
+const resolvedObjectSymbol = Symbol('ResolvedObject');
+const originalObjectSymbol = Symbol('OriginalObject');
+
+interface CreateResolvedObjectOptions {
+  contextObject?: object;
+  resolver?: ReferenceResolver;
+}
+
+export const createResolvedObject = (currentObject: object, options: CreateResolvedObjectOptions = {}): object =>
+  recursivelyCreateResolvedObject(currentObject, currentObject, [], options);
+
+const recursivelyCreateResolvedObject = (
+  currentObject: object,
+  rootCurrentObject: object,
+  propertyPath: string[],
+  options: CreateResolvedObjectOptions = {},
+): object => {
+  const mergedOptions = {
+    contextObject: options.contextObject || currentObject,
+    resolver: options.resolver || defaultResolver(options.contextObject || currentObject),
+  };
+  if (!currentObject || currentObject[resolvedObjectSymbol]) return currentObject;
+
+  const cachedValues = {};
+
+  return new Proxy(currentObject, {
+    get(target, name) {
+      if (name === resolvedObjectSymbol) return true;
```

Comment:
I found it very confusing that `obj[originalObjectSymbol]` is an `object` yet `obj[resolvedObjectSymbol]` is a boolean. Consider renaming it to `isResolvedObjectSymbol` maybe? (Also, do we suffix symbols with `Symbol`? Feels like reverse-hungarian notation a bit 😄 )

Reply:
Also the entire `resolvedObjectSymbol` is redundant. It's a resolved object exactly if it has an `[originalObjectSymbol]`, right?